### DISCUSSION TITLE
[Question] How to modify the response body after kong.response.exit(in access phase)?

### DISCUSSION THREAD

**Discussion by luozhouyang** [Kong/kong#13225] [Help] [answered]
Hi guys, I found that I can not modify the response body in a post-function plugin if kong.response.exit in access phase. What I want is to modify the response body when key-auth failed. Do you have any ideas?

**Comment by StarlightIbuki** [comment]
The API is designed so that the response cannot be overwritten. A possible workaround is to proxy the same route twice (an extra route to the proxy port) and change the response in the downstream proxy.

**Comment by luozhouyang** [comment, marked as answer]
SOLVED! I write a custom plugin to modify the response body. Here is the key steps:
* Drop the `Content-Length` header in `header_filter` phase
* Set the new body in `body_filter` phase

Here is an example:

```lua
function _M:header_filter(conf)
    kong.response.clear_header("Content-Length")
end

function _M:body_filter(conf)
    -- replace the body here
    local new_body = ...
    return kong.response.set_raw_body(new_body)
end

```

**Reply by pteric** [reply]
I've been looking for a long time how to solve it, and you have helped me a lot, thank you very much.
